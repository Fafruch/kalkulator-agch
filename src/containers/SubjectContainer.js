import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import Subject from '../components/Subjects/Subject'
import { removeSubject, toggleSubject, updateSubject } from '../actions'

const mapDispatchToProps = (dispatch) => bindActionCreators({
  onChange: updateSubject,
  onToggleClick: toggleSubject,
  onDelete: removeSubject,
}, dispatch)

const SubjectContainer = connect(null, mapDispatchToProps)(Subject)

export default SubjectContainer
