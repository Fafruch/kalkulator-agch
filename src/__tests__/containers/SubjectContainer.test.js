import React from 'react'
import { createStore } from 'redux'
import { shallow } from 'enzyme'

import rootReducer from '../../store/reducers'
import SubjectContainer from '../../containers/SubjectContainer'

const setup = (setupProps = {}) => {
  const defaultProps = {
    subject: {
      id: '0',
      name: 'Matematyka',
      elementaryScore: 100,
      advancedScore: 82,
      maxScore: {
        value: 182,
        formula: 'A/B'
      },
      active: true
    },
    subjectsType: 'primary',
    subjectNumber: 0,
    store: createStore(rootReducer)
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<SubjectContainer {...props} />)

  return {
    props,
    wrapper
  }
}

describe('SubjectContainer component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
})
