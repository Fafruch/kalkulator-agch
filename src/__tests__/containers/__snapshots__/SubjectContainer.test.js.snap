// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubjectContainer component should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Subject
    onChange={[Function]}
    onDelete={[Function]}
    onToggleClick={[Function]}
    store={
        Object {
            "asyncReducers": Object {},
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    subject={
        Object {
            "active": true,
            "advancedScore": 82,
            "elementaryScore": 100,
            "id": "0",
            "maxScore": Object {
              "formula": "A/B",
              "value": 182,
            },
            "name": "Matematyka",
          }
    }
    subjectNumber={0}
    subjectsType="primary"
/>,
  "nodes": Array [
    <Subject
      onChange={[Function]}
      onDelete={[Function]}
      onToggleClick={[Function]}
      store={
            Object {
                  "asyncReducers": Object {},
                  "dispatch": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                  Symbol(observable): [Function],
                }
      }
      subject={
            Object {
                  "active": true,
                  "advancedScore": 82,
                  "elementaryScore": 100,
                  "id": "0",
                  "maxScore": Object {
                    "formula": "A/B",
                    "value": 182,
                  },
                  "name": "Matematyka",
                }
      }
      subjectNumber={0}
      subjectsType="primary"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(Subject)
        store={
                Object {
                        "asyncReducers": Object {},
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
        subject={
                Object {
                        "active": true,
                        "advancedScore": 82,
                        "elementaryScore": 100,
                        "id": "0",
                        "maxScore": Object {
                          "formula": "A/B",
                          "value": 182,
                        },
                        "name": "Matematyka",
                      }
        }
        subjectNumber={0}
        subjectsType="primary"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
          "storeSubscription": undefined,
        },
        "props": Object {
          "store": Object {
            "asyncReducers": Object {},
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "subject": Object {
            "active": true,
            "advancedScore": 82,
            "elementaryScore": 100,
            "id": "0",
            "maxScore": Object {
              "formula": "A/B",
              "value": 182,
            },
            "name": "Matematyka",
          },
          "subjectNumber": 0,
          "subjectsType": "primary",
        },
        "propsMode": true,
        "refs": Object {},
        "renderCount": 0,
        "selector": Object {
          "error": null,
          "props": Object {
            "onChange": [Function],
            "onDelete": [Function],
            "onToggleClick": [Function],
            "store": Object {
              "asyncReducers": Object {},
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "subject": Object {
              "active": true,
              "advancedScore": 82,
              "elementaryScore": 100,
              "id": "0",
              "maxScore": Object {
                "formula": "A/B",
                "value": 182,
              },
              "name": "Matematyka",
            },
            "subjectNumber": 0,
            "subjectsType": "primary",
          },
          "run": [Function],
          "shouldComponentUpdate": false,
        },
        "setWrappedInstance": [Function],
        "state": Object {},
        "store": Object {
          "asyncReducers": Object {},
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 0,
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Subject
          onChange={[Function]}
          onDelete={[Function]}
          onToggleClick={[Function]}
          store={
                    Object {
                              "asyncReducers": Object {},
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
          subject={
                    Object {
                              "active": true,
                              "advancedScore": 82,
                              "elementaryScore": 100,
                              "id": "0",
                              "maxScore": Object {
                                "formula": "A/B",
                                "value": 182,
                              },
                              "name": "Matematyka",
                            }
          }
          subjectNumber={0}
          subjectsType="primary"
/>,
        "_debugID": 2,
        "_renderedOutput": <Subject
          onChange={[Function]}
          onDelete={[Function]}
          onToggleClick={[Function]}
          store={
                    Object {
                              "asyncReducers": Object {},
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
          subject={
                    Object {
                              "active": true,
                              "advancedScore": 82,
                              "elementaryScore": 100,
                              "id": "0",
                              "maxScore": Object {
                                "formula": "A/B",
                                "value": 182,
                              },
                              "name": "Matematyka",
                            }
          }
          subjectNumber={0}
          subjectsType="primary"
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(Subject)
    store={
        Object {
            "asyncReducers": Object {},
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    subject={
        Object {
            "active": true,
            "advancedScore": 82,
            "elementaryScore": 100,
            "id": "0",
            "maxScore": Object {
              "formula": "A/B",
              "value": 182,
            },
            "name": "Matematyka",
          }
    }
    subjectNumber={0}
    subjectsType="primary"
/>,
}
`;
